### ISSUE TITLE
Add a $merge command to be able to merge already identified users

### ISSUE THREAD

**Issue by tiina303** [PostHog/posthog#14093]
context: https://github.com/PostHog/product-internal/pull/432

Add a `$merge` command
  - longer term plan, we'll be finishing off PoE first, for now we'll create an issue to gauge interest/understand priority better.
  - restrict it more avoid attacks, accidental merges
    - requires a private api key or a special merges only api key
  - note that anything you can do with data will ideally go through the ingestion flow (there can be a button on the UI additionally), e.g. updating properties in the UI just sends a `$set` request. I'd like to keep this design principle.
    - **Similarly split IDs should be going through the ingestion flow rather than special functionality in the UI.**

**Comment by tiina303** [comment]
related issue: https://github.com/PostHog/posthog/issues/1350 (we'd need the $merge cmd to be able to provide that in the UI as we don't want to have options to do anything in the UI that isn't an ingestion event).

**Comment by NickPuljic** [comment]
I was sent here by PostHog Customer Support to add my use case and watch this issue:

We have users identified by their unique id from our Authentication Provider (AWS Cognito sub ID), and we have performed a few user modifications that require recreating the user in Cognito, generating new IDs. After these users are recreated, we associated them to our new Sub ID and create a PostHog user (using identify) - after that point we are unable to use alias to soft merge these identifiers (we didn't know this previously). 

A merge functionality would be greatly appreciated, thanks!
